### DISCUSSION TITLE
A more separated, session based history

### DISCUSSION THREAD

**Discussion by bentxt** [magicant/yash#57] [Ideas]
I usually use the Fish Shell, but I would really like to switch to yash. The thing that keeps me on the Fish Shell is how its history is a bit more separated among the different sessions. It is a really helpful feature especially if you have a lot of sessions open. 

The problem: yash puts every command in a global history file, that makes it hard to get the last commands in a session, because you have to search through non related command if you had done something in another session

The solution: keep a history file for each session

**Comment by unrealapex** [comment]
A solution to this problem might be having each Yash process generate a unique id and prepending that to HISTFILE. Each Yash process would only show history from lines with its session's id.
